### DISCUSSION TITLE
[5.x] Enhance template cache for more control over expiry

### DISCUSSION THREAD

**Discussion by rob-c-baker** [craftcms/cms#15970] [Ideas] [answered]
_TLDR; The ability to supply template cache duration in a variable and / or apply jitter to the cache duration._

In a cached template tag like this:

```twig
{% cache globally using key "[cache-key]" for 1 week %}
    ...
{% endcache %}
```

It is not possible to use a variable to supply duration. If a variable is supplied we get a Twig exception about a number being expected. It seems it's either the hard-coded duration or the default specified in general config.

It would be useful to be able to supply a duration in a variable, as we have a service / config in a module that controls different cache durations for different scenarios and adds jitter to help prevent cache stampede (at the moment, in other places we hit caches directly).

It might also be useful to allow jitter to be applied to cache durations, which should probably defined in general config, perhaps by a callback or via an Event.

**Comment by brandonkelly** [comment, marked as answer]
Just made this change for Craft 5.5 (6915d473811f29c3a9e9f52d0bccf02d59489f27).

Going forward the number can be set to any valid expression, including variable names. (The unit must still be hard-coded, e.g. `week`.)

**Reply by brandonkelly** [reply]
Craft 5.5.0 is out with that change 🎉
